src/common/darktable.h

```diff
@@ -261,7 +263,10 @@ typedef struct darktable_t
   struct dt_undo_t *undo;
   struct dt_colorspaces_t *color_profiles;
   struct dt_l10n_t *l10n;
-  dt_pthread_mutex_t db_insert;
+  dt_pthread_mutex_t db_image[DT_IMAGE_DBLOCKS];
+  pthread_mutexattr_t db_image_attr[DT_IMAGE_DBLOCKS];  
```

Comment:
I think we should avoid this declaration. Have a single pthread_mutexattr_t var and reuse it for all mutex. After all we are using the same attribute for all mutex.